astropy/wcs/base_low_level_api.py

```diff
@@ -0,0 +1,248 @@
+          coordinates.
+
+        * The last tuple element must be a dictionary with the keyword
+          arguments required to initialize the class.
+
+        See below for an example of this property. Note that we don't
+        require the classes to be Astropy classes since there is no
+        guarantee that Astropy will have all the classes to represent all
+        kinds of world coordinates. Furthermore, we recommend that the
+        output be kept as human-readable as possible.
+
+        The classes used here should have the ability to do conversions by
+        passing an instance as the first argument to the same class with
+        different arguments (e.g. ``Time(Time(...), scale='tai')``). This is
+        a requirement for the implementation of the high-level interface.
+
+        The second and third tuple elements for each value of this dictionary
+        can in turn contain either instances of classes, or if necessary can
+        contain serialized versions that should take the same form as the main
+        classes described above (a tuple with three elements with the fully
+        qualified name of the class, then the positional arguments and the
+        keyword arguments). For low-level API objects implemented in Python, we
+        recommend simply returning the actual objects (not the serialized form)
+        for optimal performance. Implementations should either always or never
+        use serialized classes to represent Python objects, and should indicate
+        which of these they follow using the ``serialized_classes`` attribute.
+
+        See the document
+        *APE 14: A shared Python interface for World Coordinate Systems*
+        for more examples (https://doi.org/10.5281/zenodo.1188875).
+        """
+        raise NotImplementedError()
+
+    # The following three properties have default fallback implementations, so
+    # they are not abstract.
+
+    @property
+    def pixel_shape(self):
+        """
+        The shape of the data that the WCS applies to as a tuple of
+        length ``pixel_n_dim``.
+
+        If the WCS is valid in the context of a dataset with a particular
+        shape, then this property can be used to store the shape of the
+        data. This can be used for example if implementing slicing of WCS
+        objects. This is an optional property, and it should return `None`
+        if a shape is not known or relevant.
+        """
+        return None
+
+    @property
+    def pixel_bounds(self):
+        """
+        The bounds (in pixel coordinates) inside which the WCS is defined,
+        as a list with ``pixel_n_dim`` ``(min, max)`` tuples.
+
+        WCS solutions are sometimes only guaranteed to be accurate within a
+        certain range of pixel values, for example when definining a WCS
+        that includes fitted distortions. This is an optional property, and
+        it should return `None` if a shape is not known or relevant.
```

Comment:
should not be 'shape' here? Should be something about pixel bounds?

Reply:
Fixed